setup.py

```diff
@@ -395,6 +395,11 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
 pagerduty = [
     'pdpyras>=4.1.2,<5',
 ]
+pandas = [
+    # Pandas stopped releasing 3.6 binaries for 1.2.* series.
+    'pandas>=0.17.1, <1.2;python_version<"3.7"',
+    'pandas>=0.17.1, <2.0;python_version>="3.7"',
```

Comment:
```suggestion
    'pandas>=0.17.1, <2.0',
```

Pandas 1.2+ provides the `Requires-Python` metadata, so 1.2+ versions will be automatically excluded when `pip install` on Python 3.6:

```
$ python3.6 -m venv x
$ . x/bin/activate
(x) $ pip install pandas
Collecting pandas
  Using cached https://.../pandas-1.1.5-cp36-cp36m-manylinux1_x86_64.whl
Collecting python-dateutil>=2.7.3 (from pandas)
...
```

So the `python_version<"3.7"` line is not needed.

Note: Python 3.6.0 ships with pip 9.1 by default, which supports `Requires-Python`, so it’s highly unlikely someone on 3.6 running on pip version not supporting this metadata.

Reply:
Good point, updated